src/components/VioletEnroll/VioletEnroll.tsx

```diff
@@ -0,0 +1,200 @@
+import { useState } from 'react'
+import { ArrowRight } from 'react-feather'
+import { Text } from 'rebass'
+import { useToggleWalletModal } from 'state/application/hooks'
+import styled from 'styled-components/macro'
+import { ExternalLink } from 'theme'
+import { logErrorWithNewRelic } from 'utils/newRelicErrorIngestion'
+
+const Wrapper = styled.div`
+  width: 100%;
+  position: relative;
+  display: flex;
+  flex-flow: column;
+  align-items: center;
+`
+
+const Container = styled.div`
+  width: 100%;
+  padding: 32px 32px 25px 32px;
+  display: flex;
+  flex-flow: column;
+  align-items: center;
+`
+
+const Label = styled.div<{ color: string; backgroundColor: string }>`
+  width: 100%;
+  padding: 5px 5px 15px;
+  background-color: ${({ backgroundColor }) => backgroundColor};
+  border-radius: 16px;
+  color: ${({ color }) => color};
+`
+
+const TitleRow = styled.div`
+  align-items: center;
+  font-weight: 700;
+  display: inline-flex;
+`
+
+const Title = styled(Text)`
+  font-weight: 600;
+  font-size: 18px;
+  line-height: 24px;
+  margin-left: 7px;
+`
+
+const DetailsRow = styled.div`
+  margin-top: 8px;
+  font-size: 12px;
+  line-height: 16px;
+  color: ${({ theme }) => theme.textSecondary};
+`
+
+const StyledLink = styled(ExternalLink)`
+  color: ${({ theme }) => theme.textSecondary};
+  font-weight: 700;
+`
+
+const ShortColumn = styled(AutoColumn)`
+  margin-bottom: 10px;
+`
+
+const StyledButton = styled(ButtonPrimary)`
+  margin-top: 24px;
+  width: 100%;
+  font-weight: 600;
+`
+
+const StyledCloseButton = styled(StyledButton)`
+  background-color: ${({ theme }) => theme.backgroundInteractive};
+  color: ${({ theme }) => theme.textPrimary};
+
+  &:hover {
+    background-color: ${({ theme }) => theme.backgroundInteractive};
+    opacity: ${({ theme }) => theme.opacity.hover};
+    transition: opacity 250ms ease;
+  }
+  padding: 8px;
+`
+
+const RegisterButton = styled(VioletProtected(ButtonPrimary, 'backgroundContrast', '0.2rem'))`
+  padding: 10px;
+`
+
+const ConnectButton = styled(ButtonPrimary)`
+  font-size: 16px;
+  font-weight: 600;
+`
+
+export default function VioletEnroll({
+  onClose,
+  keepModalOpen,
+}: {
+  onClose: () => void
+  keepModalOpen: (open: boolean) => void
+}) {
+  const { account, chainId } = useWeb3React()
+  const [registering, setRegistering] = useState(false)
+  const [enrollmentResult, setEnrollmentResult] = useState<string>('')
+  const toggleWalletModal = useToggleWalletModal()
+
+  const onEnroll = async () => {
+    setRegistering(true)
+    keepModalOpen(true)
+
+    const violetEnrollResult = await getVioletEnrollmentCall({
+      account,
+      chainId,
+    })
+
+    if (!violetEnrollResult) {
+      console.error(`Failed to enroll user`)
+      logErrorWithNewRelic({ errorString: 'Failed to enroll user' })
+      return
+    }
+
+    setEnrollmentResult(violetEnrollResult)
+    setRegistering(false)
+  }
+
+  return (
+    <Wrapper>
+      <Container>
+        {enrollmentResult ? (
```

Comment:
If this is being used as a Boolean, should we change to 'enrollmentSucceeded' or something similar? Also what is the content of this returned string?

Reply:
This is actually returning us the `tx_id` of the enrollment. I left it just in case it was useful for us later but probably makes sense to clarify this condition as something like `enrollmentResult != ''` or something?

Reply:
Fixed in https://github.com/violetprotocol/mauve-dex/pull/153/commits/74d9a4296be6e335f0cd28865421adb310b5beef